### PR TITLE
add option to return prompt

### PR DESCRIPTION
**Pull request by hwchase17**
can be useful to have access to the prompt outside of llmchain, similar to return_intermediate steps

### CODE REVIEW

**Comment by hwchase17** [comment]
should rename key to prompt

File: langchain/chains/flare/base.py [hunk 1]
```diff
@@ -36,7 +36,7 @@ def generate_tokens_and_log_probs(
         *,
         run_manager: Optional[CallbackManagerForChainRun] = None,
     ) -> Tuple[Sequence[str], Sequence[float]]:
-        llm_result = self.generate([_input], run_manager=run_manager)
+        _, llm_result = self.generate([_input], run_manager=run_manager)
```
Comment (eyurtsev) [inline langchain/chains/flare/base.py:39]: Is this PR causing a breaking change? 

If so, it'll cause a breaking change for any user that implemented anything by inheriting from LLMChain

**Review by eyurtsev — COMMENTED** [review]
Looks like a dangerous change

**Comment by dev2049** [comment]
yea not sure about this one. could we add prompt val to LLMResult or Generate?

File: langchain/chains/llm.py [hunk 2]
```diff
@@ -59,37 +60,45 @@ def output_keys(self) -> List[str]:
 
         :meta private:
         """
-        return [self.output_key]
+        if self.return_formatted_prompt:
+            return [self.output_key, "formatted_prompt"]
```
[the remark this answers is not in the document]
Comment (hwchase17) [reply langchain/chains/llm.py:64]: honestly prompt also works

**Pull request closed without merging**